### PR TITLE
build: do not run duplicated steps during jar building

### LINKED ISSUE #42714: cljs files are built twice at CI during build

### PR DESCRIPTION
**Pull request by uladzimirdev**
closes https://github.com/metabase/metabase/issues/42714

### Description

To be added

Screenshot 2024-05-15 at 17 33 25

cuts ~1min from the build

### CODE REVIEW

File: e2e/test/scenarios/dashboard-filters/parameters.cy.spec.js [hunk 1]
```diff
@@ -86,9 +86,7 @@ describe("scenarios > dashboard > parameters", () => {
     });
     selectDashboardFilter(getDashboardCard(1), "Category");
 
-    // finish editing filter and save dashboard
-    // eslint-disable-next-line no-unscoped-text-selectors -- deprecated usage
-    cy.contains("Save").click();
+    saveDashboard();
```
Comment (nemanjaglumac) [inline e2e/test/scenarios/dashboard-filters/parameters.cy.spec.js:89]: Can we please get rid of this change, and only focus on the build scripts?

**Review by nemanjaglumac — APPROVED** [review]
@uladzimirdev Left a comment about an extra diff which should be removed, imo. But other than that - this looks like it is good to go.

I was skeptical of removing the extra `yarn &&`, but that affects only local usage when people change branches. I don't think anyone will run `build-release` in that scenario :D

Apart from saving a minute in the CI, this PR will declutter the logs, which is another bonus point. 🥇 

Thank you

File: e2e/test/scenarios/dashboard-filters/parameters.cy.spec.js (hunk 1 above)
Comment (uladzimirdev) [reply e2e/test/scenarios/dashboard-filters/parameters.cy.spec.js:89]: sure, I don't know any other way to trigger e2e tests

Comment (nemanjaglumac) [reply e2e/test/scenarios/dashboard-filters/parameters.cy.spec.js:89]: Fair enough.
Let's merge it as is, then.

**Review by iethree — APPROVED** [review]
🥳

**Pull request merged**